releasetool/data/roles/upload_rpm_signatures/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: find rpms to upload signatures for
+  find:
+    path: "{{ work_dir_rpm }}"
+    patterns: "*.rpm"
+  register: rpms_to_sign
+
+- name: upload signatures
+  command: "koji import-sig {{ item.path }}"
+  loop: "{{ rpms_to_sign.files }}"
+
+- name: find srpms
+  find:
+    path: "{{ work_dir_rpm }}"
+    patterns: "*.src.rpm"
+  register: srpms
+
+- name: upload signatures
+  command: "koji write-signed-rpm {{ gpg_id }} {{ item.path | basename | replace('.src.rpm', '') }}"
```

Comment:
I think you can use a single command with many arguments. This is probably faster because koji can keep open the HTTP connection.

Reply:
I've implemented batching now. Without xargs, but see yourself :)